### ISSUE TITLE
dead code in DiagsUpdateLocList

### ISSUE THREAD

**Issue by Shane-XB-Qian** [yegappan/lsp#345]
```vim
  var LspQfId: number = bnr->getbufvar('LspQfId', 0)
  if LspQfId->empty() && !opt.lspOptions.autoPopulateDiags && !calledByCmd
    # If a location list for the diagnostics was not opened previously,
    # and 'autoPopulateDiags' is set to false, then do nothing.
    return false
  endif
```

1, i do not understand this `LspQfId->empty()`, when this would happen? seems always false?
2, and `calledByCmd` why here required to limit that, e.g after just modified one err code, it should not update diag loc list?

**Comment by yegappan** [comment]
1. The `LspQfId` buffer-local variable will not be present, if the diags location list was not previously opened using the `:LspDiagShow` command.  In this case, `LspQfId` will be 0.  Instead of using the empty() function, we can check against a non-zero value.  I have updated the comparison in 72448b8452674c871d3c85c61ecee2ffa9a1053d.
2. If the diags location list is not present, then there is no need to create a new one (unless the `autoPopulateDiags` option is set).  But if this function is invoked by the `:LspDiagShow` command, then the location list should be created.  I have updated the comments to describe this.
